feat!: support setting error messages with DatePickerI18n

vursen opened the pull request:
## Description

The PR introduces DatePickerI18n with methods for setting error messages and updates the DatePicker's validation logic to show these error messages when validation fails.

> [!WARNING]
> The PR is marked as breaking because it causes the component to start overriding any custom error messages that were set manually with the `setErrorMessage` method before validation. If you need to set error messages manually, use `setManualValidation(true)`, which prevents the component from updating the `invalid` and `errorMessage` properties, allowing you to control them manually.

Part of #4618 

## Type of change

- [x] Feature

DiegoCardoso commented on vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java:
```diff
@@ -846,6 +864,26 @@ public Registration addInvalidChangeListener(
         return addListener(InvalidChangeEvent.class, listener);
     }
 
+    private String getBadInputErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(DatePickerI18n::getBadInputErrorMessage).orElse("");
+    }
+
+    private String getRequiredErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(DatePickerI18n::getRequiredErrorMessage).orElse("");
+    }
+
+    private String getMinErrorMessage() {
+        return Optional.ofNullable(i18n).map(DatePickerI18n::getMinErrorMessage)
+                .orElse("");
+    }
+
+    private String getMaxErrorMessage() {
+        return Optional.ofNullable(i18n).map(DatePickerI18n::getMaxErrorMessage)
+                .orElse("");
+    }
```
Could use the same approach suggested in https://github.com/vaadin/flow-components/pull/6365#discussion_r1675358980

vursen replied:
Thanks, updated.

vursen commented:
During our internal discussion, we discovered a way to keep error messages that are set with `setErrorMessage`. They now take priority over i18n error messages, which allows developers to continue using the old familiar method, setting a single error message for all constraints. I've updated the PR and also removed the breaking change mark.
